### PR TITLE
[CODEGEN][CUDA]: fix cuda half math function is undefined: herf

### PR DESCRIPTION
**Pull request by kongroo**
I got an "identifier herf undefined" error when converting a model into TVM, which can be reproduced by the following code
```
import torch
import numpy as np
import tvm
from tvm import relay

class ErfTest(torch.nn.Module):
    def forward(self, data):
        return torch.erf(data)

def run(data):
    # convert pytorch to tvm
    traced = torch.jit.trace(ErfTest(), torch.from_numpy(data).cuda())
    mod, params = relay.frontend.from_pytorch(traced, [('data', data.shape)])

    # compile
    with tvm.transform.PassContext(opt_level=3):
        relay_graph, relay_lib, relay_params = tvm.relay.build(mod, target='cuda', params=params)
        relay_model = tvm.contrib.graph_runtime.create(relay_graph, relay_lib, tvm.context('gpu', 0))

    # run
    relay_model.set_input('data', data)
    relay_model.run()
    return relay_model.get_output(0)

data = np.random.rand(3, 4).astype(np.float32)
print(torch.erf(torch.Tensor(data)))
print(run(data))
print(run(data.astype(np.float16)))
```
image

I fixed this in the same way as #6225 and the results seem ok

cc @vinx13 Could you help to review this?

### CODE REVIEW

**Comment by kongroo** [comment]
> Could you add test cases for half math of erf to verify the accuracy?

I have added a test case for half erf, but ci failed. It seems that this condition `#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 530)` is not met in the ci environment, and herf is still undefined. Should I just remove the test case?

**Comment by vinx13** [comment]
You can add a check of cuda compute capability and skip the test if not supported

**Comment by tqchen** [comment]
@kongroo @vinx13 please followup
